Feat: Support for 1-arg methods in constructAnySetter()

AdamFR00 opened the pull request:
Adds support for 1-arg methods in the constructAnySetter. If constructAnySetter is called with a 1-arg method it will return an instance of MapMethodAnyProperty.  The method currently does not check the types of the arguments as it assumes it has been done in previous steps. 


Closes #8

alixpeigue committed:
Feat: add MapMethodAnyProperty, add parsingDone() public method to AnyProperty

AdamFR00 committed:
Feat: Modify constructAnySetter

constructAnySetter now supports methods with one parameter. The
implementation assumes that the class of the parameter is previously
checked. Added try catch block when parsingDone() is called.

anass-coder commented on src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializer.java:
```diff
@@ -405,6 +405,14 @@ else if (_objectIdReader != null && p.hasTokenId(JsonTokenId.ID_END_OBJECT)) {
+                try{
+                    _anySetter.parsingDone(bean);
+                }catch(Exception e){
+                    wrapAndThrow(e, bean, propName, ctxt);
```
Hey, can you add a simple comment above this block? Something like:
// Finalize extra property processing: if an any-setter exists, call parsingDone(bean)
// to pass all accumulated unmapped properties (especially for the new Map-based setter),
// and catch any errors.

anass-coder commented on src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializerFactory.java:
```diff
@@ -842,18 +842,31 @@ protected SettableAnyProperty constructAnySetter(DeserializationContext ctxt,
+            // It can either be a 2-arg method or 1-arg of type Map. 
+            if(am.getParameterCount() == 1){
+                JavaType paramType = am.getParameterType(0);
+                paramType = resolveMemberAndTypeAnnotations(ctxt, am, paramType);
```
a validation check should be added to ensure its parameter is Map-like, so we can provide a clear error if it's not.

AdamFR00 replied:
From what I understand this is checked in the BasicBeanDescription.java so that is why I did not include another check here.

anass-coder commented on src/main/java/com/fasterxml/jackson/databind/deser/SettableAnyProperty.java:
```diff
@@ -340,6 +355,32 @@ public void handleResolvedForwardReference(Object id, Object value)
+     * @since 2.19
+     */
+    protected static class MapMethodAnyProperty extends SettableAnyProperty implements java.io.Serializable {
+        Map<Object, Object> map;
```
this map should be initialized

AdamFR00 committed:
Fix: Requested changes in PR

Initialized map in MapMethodAnyProperty to avoid NullPointerException in
_set method. Added comment when for clarity when above the call to
parsingDone(bean). Did not add a check if the arg is of type map as that
is done in BasicBeanDescription.
